crates/typst-library/src/layout/grid/resolve.rs

```diff
@@ -733,6 +736,13 @@ impl<'a> CellGrid<'a> {
         // Therefore, we use a counter, 'auto_index', to determine the position
         // of the next cell with (x: auto, y: auto). It is only stepped when
         // a cell with (x: auto, y: auto), usually the vast majority, is found.
+        //
```

Comment:
as discussed, it would be nice to attempt to extract one (or maybe multiple) functions here as the `resolve` function is really large